File: dotnet/Trinsic/Storage/MemoryTokenProvider.cs
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace Trinsic;

public class MemoryTokenProvider : ITokenProvider
{
    public static MemoryTokenProvider StaticInstance => new();
    
    private Dictionary<string, string> tokens = new();
    public Task<string?> GetAsync(string name = TokenDefaults.Name, CancellationToken cancellationToken = default) {
        return Task.FromResult(tokens[name])!;
    }
    public string? Get(string name = TokenDefaults.Name) {
        return tokens[name];
    }
    public Task SaveAsync(string authToken, string name = TokenDefaults.Name, CancellationToken cancellationToken = default) {
        tokens[name] = authToken;
        return Task.CompletedTask;
    }
    public void Save(string authToken, string name = TokenDefaults.Name) {
        tokens[name] = authToken;
    }
}

File: java/src/main/java/trinsic/storage/MemoryTokenProvider.java
package trinsic.storage;

import com.google.common.util.concurrent.ListenableFuture;
import java.util.HashMap;
import trinsic.TrinsicUtilities;

public class MemoryTokenProvider implements ITokenProvider {
  HashMap<String, String> tokens = new HashMap<>();

  @Override
  public ListenableFuture<String> get(String name) {
    return TrinsicUtilities.GetListenableFutureValue(tokens.get(name));
  }

  @Override
  public ListenableFuture<String> save(String accountProfile, String name) {
    tokens.put(name, accountProfile);
    return get(name);
  }

  @Override
  public ListenableFuture<String> save(String accountProfile) {
    return save(accountProfile, "default");
  }
}
